### DISCUSSION TITLE
Proposal: Bisq needs a network status indicator

### DISCUSSION THREAD

**Discussion by ghost** [bisq-network/bisq#6326] [Ideas]
tl;dr - With an easy to read `green | yellow | red` indicator, the user will be informed whether there is an issue with the P2P network connection.  The reason for wanting this is that some users have been unaware that their node was stuck in a state of not accepting inbound connections (detailed below).

![image](https://user-images.githubusercontent.com/47253594/185265031-645b3382-a2ad-4dc1-bd19-7e6717245acf.png)

This is a proposal for having a small network status indicator shown in the bottom-right footer of the Bisq app, next to the Tor icon.
Clicking on the indicator can provide more information about the status and link to [bisq.wiki](https://bisq.wiki/Network_status_indicator) for troubleshooting tips.

> **Green** - the P2P connection is healthy, as you have a mix of outbound and inbound connections.
> **Yellow** - there are outbound but no inbound connections. Bisq can operate in this state, but it is sub-optimal. This is a normal state at application startup, until peers discover and connect to your node. However if it persists for several hours it could mean that inbound connections are blocked, the result being your offers would not be reachable. (Peers would be frustrated with the response "Maker_Offline" as a result of trying to take your offer). If this condition persists for more than a few hours, clear the Tor cache and restart Bisq after clicking on: Settings -> Network Info -> Open Tor Settings -> [Delete outdated Tor files and shut down].
> **Red** - An error situation- there are no P2P network connections. This would usually indicate that your internet connection has broken. Check other network applications such as a web browser. Bisq cannot function properly in this state. To resolve the issue, see [Troubleshooting network issues](https://bisq.wiki/Troubleshooting_network_issues)

&nbsp;

There have been numerous issues in support and observable in the order book, of users unknowingly stuck in the state of not accepting inbound Tor connections:

1. @w0000000t experienced this, for several days unaware of the problem, [commenting](https://github.com/bisq-network/roles/issues/80#issuecomment-1207488808) that @burningman3 ignored his offer in the Bsq market in preference for other offers.  We troubleshooted the problem on Matrix and found that w0000000t had no inbound peers.  Clearing the Tor cache fixed it.
2. Some users have complained that their offer never gets taken - the assumption has always been that they need to make their offer attractive and be patient.  But it could have been due to a network problem - here a status indicator would help.
3. A previous attempt to solve the issue was to do a [self-ping](https://github.com/bisq-network/bisq/pull/6147) near app startup.  In case of failure, it pops up a dialog telling them to check their firewall and inbound connections, mentioning specifically a condition for Whonix users, since it appears to be a systematic [issue](https://github.com/bisq-network/bisq/issues/6142) there.  However this has confused other users and caused them to contact support.  No one likes popups that appear out of the blue, especially confusing ones.  I propose replacing the popup with a status indicator will ease the confusion.
4. @Liisachan suggested the concept of a [status indicator](https://github.com/bisq-network/bisq/pull/6147#issuecomment-1123826820) as an improvement: _"A possible idea: an `Incoming connection OK/Bad` indicator (e.g. a small green/red circle) could be added on the status bar, showing not just a warning on error, but an ok notification on success."_
5. Other apps such as electrum have a similar network status indicator in the footer bar.
6. Being unable to take offers due to "Maker is offline" frustrates users since the offer is usually top of book - see example:
 
![image](https://user-images.githubusercontent.com/47253594/185264774-5a1597e5-6452-49c7-af56-149d1cd6863f.png)

---

**NB:** I have developed the status indicator and it is being soak tested at the moment, but I wanted to write this proposal as justification and background info for those not familiar with the issues we see in support.

&nbsp;

**Comment by w0000000t** [comment]
I completely agree, this was my instant feature request when I discovered the issue with my BSQ swap offers, and given how it would be fast to implement in the hand of @jmacxx , plus sit in an otherwise empty space, all the while adding precious, unobtrusive feedback for users, there is no reason not to add this.
